tests/functional/test_investment_pathway.py

```diff
@@ -0,0 +1,476 @@
+        - 300 at first stage
+        - 0 in child A
+        - 100 in child B
+    """
+
+    database = DataBase()
+    database.add_data("N", "spillage_cost", ConstantData(10))
+    database.add_data("N", "ens_cost", ConstantData(10_000))
+
+    database.add_data(
+        "D",
+        "demand",
+        TreeData(
+            {
+                "root": ConstantData(300),
+                "childA": ConstantData(600),
+                "childB": ConstantData(600),
+            }
+        ),
+    )
+
+    database.add_data("CAND", "op_cost", ConstantData(10))
+    database.add_data("CAND", "already_installed_capa", ConstantData(200))
+    database.add_data(
+        "CAND",
+        "invest_cost",
+        TreeData(
+            {
+                "root": ConstantData(100),
+                "childA": ConstantData(50),
+                "childB": ConstantData(50),
+            }
+        ),
+    )
+    database.add_data(
+        "CAND",
+        "max_invest",
+        TreeData(
+            {
+                "root": ConstantData(400),
+                "childA": ConstantData(100),
+                "childB": ConstantData(100),
+            }
+        ),
+    )
+
+    database.add_data("BASE", "p_max", ConstantData(200))
+    database.add_data("BASE", "cost", ConstantData(5))
+
+    COUPLING_MODEL = model(
+        id="COUPLING",
+        variables=[
+            float_variable(
+                "root_CAND_delta_invest",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+            float_variable(
+                "root_CAND_invested_capa",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+            float_variable(
+                "childA_CAND_delta_invest",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+            float_variable(
+                "childA_CAND_invested_capa",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+            float_variable(
+                "childB_CAND_delta_invest",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+            float_variable(
+                "childB_CAND_invested_capa",
+                lower_bound=literal(0),
+                structure=IndexingStructure(False, False),
+                context=ProblemContext.INVESTMENT,
+            ),
+        ],
+        constraints=[
```

Comment:
This constraints linking `invested_capa` and `delta_invest` seems to be always the same, maybe a next step in the work (not in this PR obviously) could be to automatically generate those from the tree structure ?

Reply:
Indeed! Automate it would be the next step to allow this to become more user friendly anyway and less artisanal